Automattic/eslint-config-wpvip
# Add jest testing

This adds some simple testing to confirm that the rules are applying the way we think they are.

There are two files we'll want to keep up to date as we modify our rule set:

* `__fixtures__/allowed.js` -- Syntax that should be allowed according to the rule(s) affected by proposed changes (& uncovered rules)
* `__fixtures__/disallowed.js` -- Syntax that should *NOT* be allowed according to the rule(s) affected by proposed changes (& uncovered rules)

After changes & before commit, run the following to update the snapshot of jest errors found in the "disallowed" fixture:

`npm run jest --updateSnapshot`

## To Test

Confirm that the test cases in `__tests__/checkFixtures.js` are sensical and are enforced during CI.
Confirm the snapshot update procedure is clear, easy, and effective.
